tools/e2etesting/DeployPLCs.ps1

```diff
@@ -47,74 +61,57 @@ if (!$testSuffix) {
 }
 
-
-
 ## Check if KeyVault exists
 $keyVault = Get-AzKeyVault -ResourceGroupName $ResourceGroupName
 
 if ($keyVault.Count -ne 1) {
     Write-Error "keyVault could not be automatically selected in Resource Group '$($ResourceGroupName)'."    
 } 
 
+## Ensure Azure Container Instances ##
 
+$allAciNames = @()
 
-## Deploy simulated PLCs
-$prefix = "e2etesting-simulation"
-
-$plcTemplateParameters = @{
-    "numberOfSimulations" = [int]$numberOfSimulations
-    "numberOfSlowNodes" = 1000
-    "slowNodeRate" = 10
-    "slowNodeType" = "uint"
-    "numberOfFastNodes" = 1000
-    "fastNodeRate" = 1
-    "fastNodeType" = "uint"
-    "resourcesPrefix" = "$($prefix)"
-    "resourcesSuffix" = "$($testSuffix)"
+for ($i = 0; $i -lt $NumberOfSimulations; $i++) {
+    $name = "$($ResourcesPrefix)-simulation-aci-" + $i.ToString().PadLeft(3, "0") + "-" + $testSuffix
+    $allAciNames += $name
 }
 
-$plcTemplate = [System.IO.Path]::Combine($TemplateDir, "e2e.plc.simulation.json").Replace("/", "\")
-
-Write-Host "Resource Group: $($ResourceGroupName)"
-Write-Host "Number of PLCs: $($NumberOfSimulations)"
-Write-Host "Resources prefix: $($prefix)"
-Write-Host "Resources suffix: $($testSuffix)"
-Write-Host "Key Vault Name: $($keyVault.VaultName)"
-Write-Host "ARM Template: $($plcTemplate)"
+$existingACIs = Get-AzContainerGroup -ResourceGroupName $ResourceGroupName | select -ExpandProperty Name
 
-$maxAttempts = 2;
-$attempt = 0;
-$success = $false
-
-while (++$attempt -le $maxAttempts) {
-    Write-Host
-    Write-Host "Running deployment with $($plcTemplate) (Attempt #$($attempt)/$($maxAttempts))..."
-    $plcDeployment = New-AzResourceGroupDeployment -ResourceGroupName $ResourceGroupName -TemplateFile $plcTemplate -TemplateParameterObject $plcTemplateParameters
+$aciNamesToCreate = @()
 
-    if ($plcDeployment.ProvisioningState -ne "Succeeded") {
-        Write-Warning "Deployment failed: $($plcDeployment.ProvisioningState)." -ErrorAction Stop
-    }
-    else {
-        Write-Host "Deployment successful."
-        $success = $true
-        break
+$allAciNames | %{
+    if (!@($existingACIs).Contains($_)) {
+        $aciNamesToCreate += $_
     }
 }
 
-if (!$success) {
-    Write-Error "Error while deploying simulated PLCs."    
+if ($aciNamesToCreate.Length -gt 0) {
+    $script = {
+        Write-Host "Creating ACI $($_)..."
+        $aciCommand = "/bin/sh -c './opcplc --ctb --pn=50000 --autoaccept --nospikes --nodips --nopostrend --nonegtrend --nodatavalues --sph --wp=80 --sn=$($using:NumberOfSlowNodes) --sr=$($using:SlowNodeRate) --st=$($using:SlowNodeType) --fn=$($using:NumberOfFastNodes) --fr=$($using:FastNodeRate) --ft=$($using:FastNodeType) --ph=$($_).$($using:resourceGroup.Location).azurecontainer.io'"
+        $aci = New-AzContainerGroup -ResourceGroupName $using:ResourceGroupName -Name $_ -Image $using:PLCImage -OsType Linux -Command $aciCommand -Port @(50000,80) -Cpu 1 -MemoryInGB 0.5 -IpAddressType Public -DnsNameLabel $_
```

Comment:
Please add also Cpu and Memory as script parameter

Reply:
Done.